### PR TITLE
Update docs for 1.0

### PR DESCRIPTION
**Pull request by jmaxson-ms**
Related to issue #111 .

### CODE REVIEW

File: documentation/README.md [hunk 1]
```diff
@@ -21,12 +26,9 @@ The SDK is published as a NuGet package under the name [Microsoft.Performance.SD
...
 
-# Creating A Plugin
-Refer to [Creating your first plugin](Using-the-SDK/Creating-your-plugin.md)
-
 # Recommended Reading Order
 To best understand how the SDK works and how to develop SDK plugins, it is recommended to read documentation in the following order:
-1) [Architecture/Overview](./Architecture/Overview.md) to understand at a high level the various systems the SDK provides
-2) [Architecture/The Data-Processing Pipeline](./Architecture/The-Data-Processing-Pipeline.md) to understand how to systematically process data that 
+1) [Architecture/Overview](./Architecture/README.md) to understand at a high level the various systems the SDK provides
+3) [Architecture/The Data-Processing Pipeline](./Architecture/The-Data-Processing-Pipeline.md) to understand how to systematically process data that
```
Comment (jmaxson-ms) [inline documentation/README.md:32]: hrm, not sure how this was changed. i'll update.

File: documentation/Glossary.md [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
-* [SourceParser](#sourceparser)
-* [Special Columns](#special-columns)
-* [Table](#table)
-* [TableBuilder](#tablebuilder)
-* [Table Building Cycle](#table-building-cycle)
-* [TableCommand](#tablecommand)
-* [TableConfiguration](#tableconfiguration)
-* [VisibleDomainSensitiveProjection](#visibledomainsensitiveprojection)
-* [Windows Performance Analyzer](#windows-performance-analyzer)
+- [Glossary](#glossary)
```
Comment (mslukebo) [inline documentation/Glossary.md:3]: please undo :)

File: documentation/README.md [hunk 3]
```diff
@@ -4,14 +4,19 @@ These plugins can be used by performance analysis applications that use the SDK
...
+
 # Glossary
 
 For quick definitions/examples of terms and concepts, refer to the [Glossary](./Glossary.md). 
 
 # Requirements
 The following are required in order to develop an SDK plugin:
 * [NuGet](https://www.nuget.org/downloads)
-* [.NET Standard 2.0](https://dotnet.microsoft.com/download/visual-studio-sdks)
+* [.NET Standard 2.0](https://dotnet.microsoft.com/download/visual-studio-sdks) or [compatible version](https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-1-0#tabpanel_1_net-standard-2-0)
```
Comment (mslukebo) [inline documentation/README.md:19]: nit: "or _a_ compatible version"

File: documentation/Using-the-SDK/Advanced/Creating-Your-Own-DataSource.md [hunk 4]
```diff
@@ -212,4 +212,4 @@ namespace Sample
 We have seen how to create a new kind of `DataSource` an how to seamlessly
 allow a `ProcessingSource` to leverage our new `DataSource`.
 
-[Back to Advanced Topics](Overview.md)
+[Back to Advanced Topics](README.md)
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Advanced/Creating-Your-Own-DataSource.md:215]: Does this link work? It might need to be "./README.d"

File: documentation/Using-the-SDK/Building-a-table.md [hunk 5]
```diff
@@ -1,21 +1,27 @@
 # Building a Table
 
-Table construction happens by interacting with an instance of an `ITableBuilder`. Broadly, `Tables` consist of `TableConfigurations` and Columns. Columns can be further broken down into `ColumnConfigurations` and `Projections`. An `ITableBuilder` has methods to add both Columns and `TableConfigurations`.
+Table construction happens by interacting with an instance of an `ITableBuilder`. Broadly, `Tables` consist of `TableConfigurations` and Columns. Columns can be further broken down into `ColumnConfigurations` and `Projections`. An `ITableBuilder` has methods to add `TableConfigurations` and to establish a row count for the table.
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Building-a-table.md:3]: should we keep "both Columns and"?

File: documentation/Using-the-SDK/Building-a-table.md [hunk 6]
```diff
@@ -1,21 +1,27 @@
...
-  * [ColumnConfiguration](#columnconfiguration)
-  * [Projection](#projection)
-  * [Combining ColumnConfiguration and Projections](#combining-columnconfiguration-and-projections)
-* [TableConfiguration](#tableconfiguration)
-  * [ColumnRole](#columnrole)
+*Note*: Every projection in the table must support the table's established row count.
+
+Setting the table's row count (`ITableBuilder.SetRowCount`) returns an instance of `ITableBuilderWithRowCount` which has a method to add columns to the table.
+
+- [Building a Table](#building-a-table)
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Building-a-table.md:9]: Please undo 😄

File: documentation/Using-the-SDK/Building-a-table.md (hunk 5 above)
Comment (mslukebo) [reply documentation/Using-the-SDK/Building-a-table.md:3]: Wonder if this should be a list

"An `ITableBuilder has methods to
1) Add table configurations
2) Set a row count
3) Add columns

File: documentation/Using-the-SDK/Building-a-table.md [hunk 7]
```diff
@@ -1,21 +1,27 @@
...
 
-* [Column](#column)
-  * [ColumnConfiguration](#columnconfiguration)
-  * [Projection](#projection)
-  * [Combining ColumnConfiguration and Projections](#combining-columnconfiguration-and-projections)
-* [TableConfiguration](#tableconfiguration)
-  * [ColumnRole](#columnrole)
+*Note*: Every projection in the table must support the table's established row count.
+
+Setting the table's row count (`ITableBuilder.SetRowCount`) returns an instance of `ITableBuilderWithRowCount` which has a method to add columns to the table.
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Building-a-table.md:7]: Maybe this should be its own section below?

File: documentation/Using-the-SDK/Building-a-table.md [hunk 8]
```diff
@@ -1,21 +1,27 @@
...
-Table construction happens by interacting with an instance of an `ITableBuilder`. Broadly, `Tables` consist of `TableConfigurations` and Columns. Columns can be further broken down into `ColumnConfigurations` and `Projections`. An `ITableBuilder` has methods to add both Columns and `TableConfigurations`.
+Table construction happens by interacting with an instance of an `ITableBuilder`. Broadly, `Tables` consist of `TableConfigurations` and Columns. Columns can be further broken down into `ColumnConfigurations` and `Projections`. An `ITableBuilder` has methods to add `TableConfigurations` and to establish a row count for the table.
 
-* [Column](#column)
-  * [ColumnConfiguration](#columnconfiguration)
-  * [Projection](#projection)
-  * [Combining ColumnConfiguration and Projections](#combining-columnconfiguration-and-projections)
-* [TableConfiguration](#tableconfiguration)
-  * [ColumnRole](#columnrole)
+*Note*: Every projection in the table must support the table's established row count.
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Building-a-table.md:5]: I think it might make sense to move this note in the "Adding columns" section.

Also, we should make it more in-your-face:

> ❗ Every projection in the table must support the table's established row count

File: documentation/Using-the-SDK/Building-a-table.md [hunk 9]
```diff
@@ -1,21 +1,27 @@
...
+    - [Projection](#projection)
+    - [Combining ColumnConfiguration and Projections](#combining-columnconfiguration-and-projections)
+  - [TableConfiguration](#tableconfiguration)
+    - [ColumnRole](#columnrole)
+- [More Information](#more-information)
 
 ## Column
 
-A column is a conceptual (`ColumnConfiguration`, `Projection`) pair that defines data inside a [Table](#table).
+A column is a conceptual pair (`ColumnConfiguration`, `Projection`) that defines data inside a [Table](#table).
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Building-a-table.md:20]: Can you remove the Table link? It was copied from the glossary by mistake.

File: documentation/Using-the-SDK/Creating-a-table.md [hunk 10]
```diff
@@ -9,8 +9,12 @@ including cookers that may not necessarily be shipped with said `Table`, to buil
 exposes data through cookers, then you can author a table to leverage said data.
 
 Creating a table involves two keys steps:
-* [Declaring the Table](#declaring-the-table)
-* [Integrating the Table with our Data-Processing Pipeline](#integrating-the-table-with-our-data-processing-pipeline)
+- [Creating a Table](#creating-a-table)
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Creating-a-table.md:12]: Please undo 😄

File: documentation/Using-the-SDK/Creating-a-table.md [hunk 11]
```diff
@@ -9,8 +9,12 @@ including cookers that may not necessarily be shipped with said `Table`, to buil
...
+  - [Declaring the Table](#declaring-the-table)
+  - [Integrating the Table with our Data-Processing Pipeline](#integrating-the-table-with-our-data-processing-pipeline)
+- [Examples](#examples)
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Creating-a-table.md:15]: Do we want these links?

File: documentation/Using-the-SDK/Creating-your-plugin.md [hunk 12]
```diff
@@ -4,33 +4,34 @@ This document outlines how to use the Performance Toolkit SDK (SDK) to create
 an SDK plugin. Plugins can be used to expose data from `DataSource`s for
 automation, trace extractors, or viewers (e.g. Windows Performance Analyzer).
 
-Before creating a plugin, it is recommended to read [the overview of the SDK's architecture](../Architecture/Overview.md).
+Before creating a plugin, it is recommended to read [the overview of the SDK's architecture](../Architecture/README.md).
 
 Creating a plugin can be outlined into 4 distinct steps:
-1. [Creating the Project](#creating-the-project)
+- [Creating an SDK Plugin](#creating-an-sdk-plugin)
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Creating-your-plugin.md:10]: Same here

File: documentation/Using-the-SDK/Creating-your-plugin.md [hunk 13]
```diff
@@ -4,33 +4,34 @@ This document outlines how to use the Performance Toolkit SDK (SDK) to create
...
+  - [Creating a CustomDataProcessor and Tables](#creating-a-customdataprocessor-and-tables)
 
 ---
 
 ## Creating the Project
 
 Plugins are created as C# class libraries that get dynamically loaded by the SDK runtime. To begin your plugin creation, we will first walk through creating a C# project.
 
-> :information_source: The SDK team is actively working on creating a dotnet template to simplify creating your project and writing the neccessary plugin boilerplate code. 
+*Note*: The SDK team is [actively working](https://github.com/microsoft/microsoft-performance-toolkit-sdk/issues/190) to create a dotnet template to simplify creating your project and writing the necessary plugin boilerplate code.
```
Comment (mslukebo) [inline documentation/Using-the-SDK/Creating-your-plugin.md:34]: Why the change from `> :information_source:`?

Comment (mslukebo) [inline documentation/Using-the-SDK/Creating-your-plugin.md:34]: nit: "to create" => "on creating"

Comment (mslukebo) [inline documentation/Using-the-SDK/Creating-your-plugin.md:34]: We can probably nix "the necessary" IMO

File: documentation/Using-the-SDK/Building-a-table.md (hunk 5 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Building-a-table.md:3]: no, you cannot use it to add columns.

File: documentation/Using-the-SDK/Building-a-table.md (hunk 7 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Building-a-table.md:7]: We had adding columns in this section, and it's part of building a table. I'm ok if we want to make additional changes, but based on the headings, this seemed like the right place to leave it. If you have a specific idea of additional headings, please let me know.

File: documentation/Using-the-SDK/Creating-your-plugin.md (hunk 13 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Creating-your-plugin.md:34]: It doesn't show up well in code. Is it a github specific extension to markdown?

File: documentation/Glossary.md (hunk 2 above)
Comment (jmaxson-ms) [reply documentation/Glossary.md:3]: stupid auto updates

File: documentation/Using-the-SDK/Creating-your-plugin.md (hunk 13 above)
Comment (mslukebo) [reply documentation/Using-the-SDK/Creating-your-plugin.md:34]: Yep, it looks like this:
image

Comment (mslukebo) [reply documentation/Using-the-SDK/Creating-your-plugin.md:34]: the `:emoji:` is GitHub specified, `>` is the general markdown for block quotes

File: documentation/Using-the-SDK/Creating-a-table.md (hunk 11 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Creating-a-table.md:15]: I don't care, they were another auto add. I'll just remove them for now.

File: documentation/Using-the-SDK/Advanced/Creating-Your-Own-DataSource.md (hunk 4 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Advanced/Creating-Your-Own-DataSource.md:215]: it works both on github and in vs code. leaving it for now

File: documentation/Using-the-SDK/Building-a-table.md (hunk 8 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Building-a-table.md:5]: sure

Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Building-a-table.md:5]: wait, what Adding columns section?

File: documentation/Using-the-SDK/Building-a-table.md (hunk 5 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Building-a-table.md:3]: to be clear, there is no ITableBuilder.AddColumn method

File: documentation/Using-the-SDK/Building-a-table.md (hunk 9 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Building-a-table.md:20]: sure

File: documentation/Using-the-SDK/Creating-your-plugin.md (hunk 13 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Creating-your-plugin.md:34]: yeah, looks like it is github specific. do we still want to use it?

Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Creating-your-plugin.md:34]: that's really wordy and awkward. i'm changing to "will be creating", lemme know if you have issues with that.

Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Creating-your-plugin.md]: ah, i'll use the unicode character and the >

File: documentation/README.md [hunk 14]
```diff
@@ -4,14 +4,19 @@ These plugins can be used by performance analysis applications that use the SDK
 
 This folder contains instructions for developing SDK plugins and utilizing the SDK's various features.
 
+# Quick Start
+Refer to [Creating your first plugin](Using-the-SDK/Creating-your-plugin.md) to jump in quickly.
+
+> ⚠️ We recommend following the [Recommended Reading Order](#recommended-reading-order).
```
Comment (anramanath) [inline documentation/README.md:10]: Wondering if it might just be better to put the recommended reading order in here and deleting that Header below.

File: documentation/README.md [hunk 15]
```diff
@@ -21,12 +26,9 @@ The SDK is published as a NuGet package under the name [Microsoft.Performance.SD
...
 the Visual Studio NuGet Package Manager, `dotnet.exe`, or `nuget.exe`.
 
-# Creating A Plugin
-Refer to [Creating your first plugin](Using-the-SDK/Creating-your-plugin.md)
-
 # Recommended Reading Order
 To best understand how the SDK works and how to develop SDK plugins, it is recommended to read documentation in the following order:
-1) [Architecture/Overview](./Architecture/Overview.md) to understand at a high level the various systems the SDK provides
-2) [Architecture/The Data-Processing Pipeline](./Architecture/The-Data-Processing-Pipeline.md) to understand how to systematically process data that 
+1. [Architecture/Overview](./Architecture/README.md) to understand at a high level the various systems the SDK provides
```
Comment (anramanath) [inline documentation/README.md:31]: Wondering if we should make the display text say "Architecture/README" if we want it to be a path or just "Overview" for a better description.

File: documentation/README.md [hunk 16]
```diff
@@ -21,12 +26,9 @@ The SDK is published as a NuGet package under the name [Microsoft.Performance.SD
...
+2. [Architecture/The Data-Processing Pipeline](./Architecture/The-Data-Processing-Pipeline.md) to understand how to systematically process data that
```
Comment (anramanath) [inline documentation/README.md:32]: same as above

File: documentation/Using-the-SDK/Creating-your-plugin.md [hunk 17]
```diff
@@ -4,33 +4,33 @@ This document outlines how to use the Performance Toolkit SDK (SDK) to create
 an SDK plugin. Plugins can be used to expose data from `DataSource`s for
 automation, trace extractors, or viewers (e.g. Windows Performance Analyzer).
 
-Before creating a plugin, it is recommended to read [the overview of the SDK's architecture](../Architecture/Overview.md).
+Before creating a plugin, it is recommended to read [the overview of the SDK's architecture](../Architecture/README.md).
 
 Creating a plugin can be outlined into 4 distinct steps:
```
Comment (anramanath) [inline documentation/Using-the-SDK/Creating-your-plugin.md:9]: Mind changing this to 4 components or something? If it's "steps" I feel we should have it be a numbered list.

File: documentation/Using-the-SDK/Creating-your-plugin.md [hunk 18]
```diff
@@ -4,33 +4,33 @@ This document outlines how to use the Performance Toolkit SDK (SDK) to create
...
+* [Creating a CustomDataProcessor and Tables](#creating-a-customdataprocessor-and-tables)
 
 ---
 
 ## Creating the Project
 
 Plugins are created as C# class libraries that get dynamically loaded by the SDK runtime. To begin your plugin creation, we will first walk through creating a C# project.
 
-> :information_source: The SDK team is actively working on creating a dotnet template to simplify creating your project and writing the neccessary plugin boilerplate code. 
+> ⚠️ The SDK team will be creating [(see issue)](https://github.com/microsoft/microsoft-performance-toolkit-sdk/issues/190) a dotnet template to simplify creating your project and writing boilerplate code.
```
Comment (anramanath) [inline documentation/Using-the-SDK/Creating-your-plugin.md:33]: Added a comment to the issue in order to make sure we alter the documentation as needed.

File: documentation/README.md (hunk 14 above)
Comment (jmaxson-ms) [reply documentation/README.md]: I like it better as two different sections as they have different reasons to be.

File: documentation/README.md (hunk 15 above)
Comment (jmaxson-ms) [reply documentation/README.md:31]: The overview describes the file contents, README.md is the file name because it will show up on github if you navigate to that folder. I don't think they need to match.

File: documentation/README.md (hunk 16 above)
Comment (jmaxson-ms) [reply documentation/README.md:32]: same response

File: documentation/Using-the-SDK/Creating-your-plugin.md (hunk 17 above)
Comment (jmaxson-ms) [reply documentation/Using-the-SDK/Creating-your-plugin.md]: I changed it to "actions."

**Review by mslukebo — APPROVED** [review]

**Review by anramanath — APPROVED** [review]
